### PR TITLE
Support different unit measures in spatial queries (#13204)

### LINKED ISSUE #13204: Allow to specify the unit distance in spatial queries
### Description

Follow up of https://issues.redhat.com/browse/ISPN-8238.

Distances are used in case of:

1. within circle predicate
2. projection
3. sorting

In all those cases we want to allow the user to change the default (the meter unit measure).

### Implementation ideas

We can add an optional parameter after the distance value to specify the unit measure.
The possible values should be defined in an enumeration.

### PR DESCRIPTION
**Pull request by fax4ever**
Closes #13204 

Draft - remaining tasks: => done!
1. ~~Support different unit measures in spatial predicates~~ done!
2. ~~Support different unit measures in spatial projections~~ done!
3. ~~Document unit measures in spatial queries~~ done!
4. ~~Improve the parsing of the units as soft keys~~ done!

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/dsl/embedded/impl/DistanceUnitHelper.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Locale;
+
+import org.hibernate.search.engine.spatial.DistanceUnit;
+
+public final class DistanceUnitHelper {
+
+   private DistanceUnitHelper() {
+   }
+
+   public static DistanceUnit distanceUnit(String value) {
```
Comment (tristantarrant) [inline query/src/main/java/org/infinispan/query/dsl/embedded/impl/DistanceUnitHelper.java:12]: You can use a switch here

Comment (fax4ever) [reply query/src/main/java/org/infinispan/query/dsl/embedded/impl/DistanceUnitHelper.java:12]: I'm applying it. Thanks

**Comment by fax4ever** [comment]
@andyuk1986 I don't know if you any suggestion for this contribution. In case, I'll be happy to address it

**Comment by tristantarrant** [comment]
Merged, thanks

**Comment by fax4ever** [comment]
Thanks Tristan!

**Pull request merged**